Document maxPhraseLength argument on @vector directive (#317)
neo4j/docs-graphql: modules/ROOT/pages/directives/indexes-and-constraints.adoc

  queryName: String
  """(Optional) The name of the provider."""
  provider: String
  """(Optional) The maximum number of characters allowed in a query phrase."""
  maxPhraseLength: Int
}
----

If the optional field `provider` is set, the type is used for a query by phrase, otherwise for a query by vector.
Allowed values for the `provider` field are defined by the available link:https://neo4j.com/docs/cypher-manual/current/genai-integrations/#ai-providers[GenAI providers].

The optional field `maxPhraseLength` limits the number of characters allowed in a query phrase.
When set, any query by phrase whose `phrase` argument is longer than this limit is rejected with an error before the phrase is forwarded to the GenAI plugin, so no embedding is generated for it.
This lets server owners cap the cost that clients can incur against their GenAI provider.
The value must be a positive integer, and length is measured in characters (Unicode code points).
Because it only applies to a query by phrase, `maxPhraseLength` can only be set on an index that also defines a `provider`; setting it on a query-by-vector index (one without a `provider`) is a schema build error.

> 

Author: ```suggestion
```

Then it returns all `Product` nodes with a vector embedding on their `descriptionVector` property which are similar to the vector embedding generated by the plugin.

===== Limiting the phrase length

Because each query by phrase forwards the phrase to a GenAI provider to generate an embedding, longer phrases can increase the cost incurred against that provider.
Set the optional `maxPhraseLength` field to cap the number of characters accepted in the `phrase` argument:

.Type definition
[source, graphql]
----
type Product @node @vector(indexes: [{
  indexName: "productDescriptionIndex",
  embeddingProperty: "descriptionVector",
  provider: OPEN_AI,  # Assuming this is configured in the server
  queryName: "searchByPhrase",
  maxPhraseLength: 100
}]) {
  id: ID!
  name: String!
  description: String!
}
----

With this configuration, a `searchByPhrase` query whose `phrase` argument is longer than 100 characters is rejected with an error before any embedding is generated, and no request is made to the GenAI provider.
Phrases of 100 characters or fewer are unaffected.

> Because it only applies to a query by phrase, `maxPhraseLength` can only be set on an index that also defines a `provider`; setting it on a query-by-vector index (one without a `provider`) is a schema build error.

Reviewer: ```suggestion
Because it only applies to a query by phrase, `maxPhraseLength` can only be set on an index that also defines a `provider`.
Setting it on a query-by-vector index (one without a `provider`) results in a schema build error.
```

First, the query passes the query phrase argument `$phrase` to the GenAI plugin and lets it generate a vector embedding for the phrase.
Then it returns all `Product` nodes with a vector embedding on their `descriptionVector` property which are similar to the vector embedding generated by the plugin.

===== Limiting the phrase length

Because each query by phrase forwards the phrase to a GenAI provider to generate an embedding, longer phrases can increase the cost incurred against that provider.
Set the optional `maxPhraseLength` field to cap the number of characters accepted in the `phrase` argument:

.Type definition
[source, graphql]
----
type Product @node @vector(indexes: [{
  indexName: "productDescriptionIndex",
  embeddingProperty: "descriptionVector",
  provider: OPEN_AI,  # Assuming this is configured in the server
  queryName: "searchByPhrase",
  maxPhraseLength: 100
}]) {
  id: ID!
  name: String!
  description: String!
}
----

With this configuration, a `searchByPhrase` query whose `phrase` argument is longer than 100 characters is rejected with an error before any embedding is generated, and no request is made to the GenAI provider.

> Phrases of 100 characters or fewer are unaffected.

Author: ```suggestion
Phrases of 100 characters or fewer are unaffected.

Because it only applies to a query by phrase, `maxPhraseLength` can only be set on an index that also defines a `provider`; setting it on a query-by-vector index (one without a `provider`) is a schema build error.
```
